mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java

```diff
@@ -1325,7 +1330,7 @@ public void validate(final MessageRecorder msgRecorder) {
                     // Is the level foreign key name a duplicate
                     if (columnsToObjects.containsKey(level.getColumnName().getName())) {
                         Level l = (Level)
-                            columnsToObjects.get(level.getColumnName());
+                            columnsToObjects.get(level.getColumnName().getName());
```

Comment:
Rename this.

getColumn().getname()

Pleae verif all uses

Reply:
agree

Reply:
done